packages/openapi-ts/src/utils/write/types.ts

```diff
@@ -184,6 +186,116 @@ const processType = (client: Client, model: Model, onNode: OnNode) => {
     onNode,
     type: toType(model),
   });
+
+  generateTransform(client, model, onNode);
+};
+
+const generateTransform = (client: Client, model: Model, onNode: OnNode) => {
+  const config = getConfig();
+  if (config.types.dates === 'types+transform') {
+    if (model.meta?.hasTransformer) {
+      // Transform already created (maybe the model is used in other models) so we just bail here
+      return;
+    }
+
+    const generateForProperty = (
+      rootPath: string[],
+      property: Model,
+      isArrayElement?: boolean,
+    ) => {
+      const localPath = [...rootPath, property.name];
+
+      if (
+        property.type === 'string' &&
+        property.export !== 'array' &&
+        (property.format === 'date-time' || property.format === 'date')
+      ) {
+        return [
+          compiler.transform.dateTransformMutation({
+            path: localPath,
+          }),
+        ];
+      } else if (isArrayElement) {
+        // Avoid infinite recursion on arrays
+        return [];
+      } else {
+        // otherwise we recurse in case it's an object/array, and if it's not that will just bail with []
+        return generateForModel(localPath, property);
+      }
+    };
+
+    function generateForArray(localPath: string[], localModel: Model) {
+      if (localModel.export !== 'array') {
+        throw new Error(
+          'generateForArray should only be called with array models',
+        );
+      }
+
+      if (localModel.$refs.length === 1) {
+        const nextModel = client.models.find(
+          (m) => m.meta!.name === localModel.type,
+        );
+        if (!nextModel) {
+          throw new Error(
+            `Model ${localModel.type} could not be founded when building array transform`,
+          );
+        }
+
+        generateTransform(client, nextModel, onNode);
```

Comment:
If we come across a model, which has keys referencing other models, we cannot generate a transformer until the child nexted references have transformers. For this reason we eagerly generate transformers down through the tree

Reply:
Risk: are recursive or 2-way referencing types possible in OpenAPI? Should we come up with a way to detect and bail on this?